metabase/metabase
# Connection impersonation FE

Epic: https://github.com/metabase/metabase/issues/29491

### Description

Updates and connects the impersonation UI to the backend.

### How to verify

There is a known bug. If you configure an impersonation for the first time, it will reset your native query editing permission from "Yes" to "No". I left a couple of [FIXME](https://github.com/metabase/metabase/pull/30867/files#diff-255dc4e7aa757a489ef9b7b41fe27e5ba89f0e1a14ab22659a4496327c59b55cR146) in the specs code.

- Run Metabase EE
- Connect postgres & snowflake databases
- Create roles in postgres and snowflake that limit access to certain tables
- Go to Admin -> People, create a new user, assign two attributes to the user: pg_role=*your_pg_role*, snowflake_role=*your_snowflake_role*
- Go to Admin -> Permissions, for "All users" set the data access to `Impersonated`, select corresponding attributes for pg and snowflake
- Login as the user
- Ensure users cannot see any data beyond what the roles give access to both for structured and native queries

### Demo

<img width="1363" alt="Screen Shot 2023-05-19 at 5 40 33 PM" src="https://github.com/metabase/metabase/assets/14301985/b3186ba7-b1b0-448b-b833-f8ef19830286">

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/30867)
